### PR TITLE
Refactor/string view

### PR DESCRIPTION
**Pull request by skdltmxn**
Refactored string_view usages through codebase

- Modified `const std::string_view&` to `std::string_view`
- Modified a number of potentially problematic usages of string_view such as printing out to exception messages

### CODE REVIEW

File: src/openrct2/core/Zip.cpp [hunk 1]
```diff
@@ -31,7 +31,7 @@ class ZipArchive final : public IZipArchive
         }
 
         int32_t error;
-        _zip = zip_open(path.data(), zipOpenMode, &error);
+        _zip = zip_open(std::string(path).data(), zipOpenMode, &error);
```
Comment (ZehMatt) [inline src/openrct2/core/Zip.cpp:34]: You should use c_str() to if you want null terminated string, I know you can argue that most implementations do same as data() but for correctness it should be c_str()

This should be changed in all cases.

File: src/openrct2/localisation/Convert.cpp [hunk 2]
```diff
@@ -85,7 +85,7 @@ static std::string DecodeToMultiByte(const std::string_view& src)
 /**
  * Encodes a UTF-8 string as an RCT2 string.
  */
-static std::string Encode(const std::string_view& src)
+static std::string Encode(const std::string& src)
```
Comment (ZehMatt) [inline src/openrct2/localisation/Convert.cpp:88]: Whats the reason behind the change here?

File: src/openrct2/scripting/Plugin.cpp [hunk 3]
```diff
@@ -155,10 +155,10 @@ PluginType Plugin::ParsePluginType(const std::string_view& type)
     throw std::invalid_argument("Unknown plugin type.");
 }
 
-void Plugin::CheckForLicence(const DukValue& dukLicence, const std::string_view& pluginName)
+void Plugin::CheckForLicence(const DukValue& dukLicence, std::string_view pluginName)
 {
     if (dukLicence.type() != DukValue::Type::STRING || dukLicence.as_string().empty())
-        log_error("Plugin %s does not specify a licence", pluginName.data());
+        log_error("Plugin %s does not specify a licence", std::string(pluginName).data());
```
Comment (ZehMatt) [inline src/openrct2/scripting/Plugin.cpp:161]:
```suggestion
        log_error("Plugin %s does not specify a licence", std::string(pluginName).c_str());
```

File: test/tests/ImageImporterTests.cpp [hunk 4]
```diff
@@ -19,7 +19,7 @@ using namespace OpenRCT2::Drawing;
 class ImageImporterTests : public testing::Test
 {
 public:
-    static std::string GetImagePath(const std::string_view& name)
+    static std::string GetImagePath(const std::string& name)
     {
         return Path::Combine(TestData::GetBasePath(), "images", name.data());
```
Comment (ZehMatt) [inline test/tests/ImageImporterTests.cpp:24]:
```suggestion
        return Path::Combine(TestData::GetBasePath(), "images", name.c_str());
```

File: src/openrct2/object/ObjectFactory.cpp [hunk 5]
```diff
@@ -52,14 +52,14 @@ class FileSystemDataRetriever : public IFileDataRetriever
...
+    FileSystemDataRetriever(std::string_view basePath)
         : _basePath(basePath)
     {
     }
 
-    std::vector<uint8_t> GetData(const std::string_view& path) const override
+    std::vector<uint8_t> GetData(std::string_view path) const override
     {
-        auto absolutePath = Path::Combine(_basePath, path.data());
+        auto absolutePath = Path::Combine(_basePath, std::string(path).data());
```
Comment (ZehMatt) [inline src/openrct2/object/ObjectFactory.cpp:62]:
```suggestion
        auto absolutePath = Path::Combine(_basePath, std::string(path).c_str());
```

**Review by ZehMatt — CHANGES_REQUESTED** [review]
Overall pretty good changes, just a few minor things here and there.

File: src/openrct2/core/Zip.cpp (hunk 1 above)
Comment (skdltmxn) [reply src/openrct2/core/Zip.cpp:34]: Actually as of C++11, c_str() and data() are exactly the same by standard ;)

File: src/openrct2/localisation/Convert.cpp (hunk 2 above)
Comment (skdltmxn) [reply src/openrct2/localisation/Convert.cpp:88]: Because `src` is treated as null-terminating string inside of `Encode` function.

File: src/openrct2/core/Zip.cpp (hunk 1 above)
Comment (ZehMatt) [reply src/openrct2/core/Zip.cpp:34]: I still don't like the mixed usage, c_str() is preferred to clarify intent, when data() is used its usually assumed that its buffer is used as raw data.

File: src/openrct2/localisation/Convert.cpp (hunk 2 above)
Comment (ZehMatt) [reply src/openrct2/localisation/Convert.cpp:88]: This encode function is not pretty but yeah I see the point, ideally it should not seek for the null terminator and rather use the range specified by string_view/string but thats for another PR.

File: src/openrct2/core/Zip.cpp (hunk 1 above)
Comment (skdltmxn) [reply src/openrct2/core/Zip.cpp:34]: Fair point there 👍 
I will fix them

File: src/openrct2/core/File.cpp [hunk 6]
```diff
@@ -120,11 +113,17 @@ namespace File
         return lines;
     }
 
+    void WriteAllBytes(const std::string& path, const void* buffer, size_t length)
```
Comment (Gymnasiast) [inline src/openrct2/core/File.cpp:116]: Why did you move this function? Makes the diff noisier than necessary.

**Review by ZehMatt — APPROVED** [review]
Looks good, I did no testing on this but I see no issues here.

File: src/openrct2/core/Zip.cpp (hunk 1 above)
Comment (IntelOrca) [reply src/openrct2/core/Zip.cpp]: When dealing with `string_view`, it deliberately does not have c_str as it might not be null terminated, you never know with `data()`.

File: src/openrct2/core/File.cpp (hunk 6 above)
Comment (skdltmxn) [reply src/openrct2/core/File.cpp:116]: Because I find it more readable to have all Read* APIs in a row and then Write API.

**Review by tupaschoal — APPROVED** [review]
Seems fine to me, looks pretty safe
